### ISSUE TITLE
[REQ] Required properties of a model should be made mandatory in constructor

### ISSUE THREAD

**Issue by mimkorn** [OpenAPITools/openapi-generator#9789]
## Problem

The constructors of the generated models with the spring generator have no builders, empty constructors. The fact that a property is required is only in an annotation that I do not think has any other effect than informing the reader. 

## Describe the solution you'd like

IMHO there should be a constructor that would have all the required properties in it and no constructor without it. This way you'd enforce setting those up. Otherwise what stops you from sending responses from the server without required properties? 

## Describe alternatives you've considered

using lombok to create builders and constructors would be nice in general.

Is there some way generated server stub ensures adhering to the contract in the question of required attribute?

**Comment by RockyMM** [comment]
Would not javax.validation take care of this req? I am not sure, I am just asking.

However, your request makes sense. 

Lombok should not be used as it is adding yet another dependency to the generated stuff, and it has it's own... drawbacks.

**Comment by OlliL** [comment]
There is no way to enforce "is required". This request is pointless imho.
- removing the default constructor makes using many maping frameworks impossible
- removing the default constructor breaks the POJO spec
- adding a constructor with only-required parameters but w/o nullcheck does not prevent you from handing in null values
- you still can set parameters to null afterwards
It might add a way of adding more convinience using that POJO by having an additional constructor - but thats all, nothing else.

To really enforce "is required" you would need to:
- make class final
- make default constructor private
- add null checks to constructor with only-required parameters
- add null checks to all setter-methods

but then you get a POJO which can't be used with mapping or serialization/deserialization frameworks.... so stop attempting something you can't reach

**Comment by OlliL** [comment]
Yeah you can do that, but you most likeley can't remove the default constructor, so there will always be a way of having null values for a required-prop. So yes - you can make the lifes of a software developer easier by having that additional constructor so when constructing this class you can anticipate what properties needs to be filled. But - on the other side you can't rely on it so you have to check if all required props are set anyway when consuming that object. It adds convenience but no assurance imho.
